src/algebra/lie/non_unital_non_assoc_algebra.lean

```diff
@@ -35,27 +42,45 @@ universes u v w
 
 variables (R : Type u) (L : Type v) [comm_ring R] [lie_ring L] [lie_algebra R L]
 
+/-- Type synonym for turning a `lie_ring` into a `non_unital_non_assoc_semiring`.
+
+A `lie_ring` can be regarded as a `non_unital_non_assoc_semiring` by turning its
+`has_bracket` (denoted `⁅, ⁆`) into a `has_mul` (denoted `*`). -/
+def commutator_ring (L : Type v) : Type v := L
+
 /-- A `lie_ring` can be regarded as a `non_unital_non_assoc_semiring` by turning its
 `has_bracket` (denoted `⁅, ⁆`) into a `has_mul` (denoted `*`). -/
-def lie_ring.to_non_unital_non_assoc_semiring : non_unital_non_assoc_semiring L :=
+instance : non_unital_non_assoc_semiring (commutator_ring L) :=
+show non_unital_non_assoc_semiring L, from
 { mul           := has_bracket.bracket,
   left_distrib  := lie_add,
   right_distrib := add_lie,
   zero_mul      := zero_lie,
   mul_zero      := lie_zero,
   .. (infer_instance : add_comm_monoid L) }
```

Comment:
can't you just `@[derive]` this instace, (and also the `lie_ring/lie_algebra` ones) or does that cause issues?

Reply:
This one definitely not (since it's new data), nor the others, because we need an e.g. `lie_ring L` hypothesis that is not included in the hypotheses of `commutator_ring`.